### PR TITLE
DB 

### PR DESCRIPTION
**Pull request by cha0sg0d**
This PR creates a `db` package specifically for our database. It houses the `schema.prisma`, the `docker-compose`, and the `migrations`. 

It is possible to have one location for these files due to the fact that monorepos share a `node_modules/` folder and prisma writes it's client to a specific location in the `node_modules/` folder:

> This command also runs the prisma generate command, which [generates the Prisma Client into the node_modules/.prisma/client directory](https://www.prisma.io/docs/concepts/components/prisma-client/working-with-prismaclient/generating-prisma-client#the-prismaclient-npm-package).

This allows use to remove our `pre-commit` check and have a single source of truth for db files. 

The only thing to remember is that when you want to run any dev commands, such as building the merkle tree, writing test data, or starting the frontend, you must run `pnpm db:start` at the top level first.

Would love feedback / discussion on pros / cons of this approach cc @amirgamil @DanTehrani.

Possible problems:
- The bundle size gets too big on Vercel
- The fact that we don't explicitly make `frontend`, `merkle_tree`, or `test_data` depend on `db` gets us in trouble. We don't _need_ to have this dependency because in the root `install` command we generate the prisma schema and it gets copied to the top level `node_modules/.prisma/client`.

### CODE REVIEW

**Comment by DanTehrani** [comment]
@cha0sg0d 

So in Render/Vercel, do we deploy the top-level node_modules folder?
Also, how should we run database migrations on Render (i.e. hosted postgres)?

I'm unsure if having a db package will work well in deployments, but I would like to know if I'm wrong on that.

(If the db package doesn't work well with deployments,  I think we can just have duplicate schema files and migration files in each package. And docker-compose.yml can be placed in the root dir.)

**Comment by cha0sg0d** [comment]
> I'm unsure if having a db package will work well in deployments, but I would like to know if I'm wrong on that.

This is a good point. I know that this is not a problem for Vercel. We are deploying successfully as demonstrated by the Vercel bot above. I don't have as much experience with Render, but I can play around with it and see. 

Based on this [doc](https://render.com/docs/monorepo-support#sample-monorepo), it appears Render does have basic monorepo support. 

What is your flow you normally do for DB deployments + migrations? This will help me too.

**Comment by DanTehrani** [comment]
@cha0sg0d 
Yeah for db migrations, I think we can run them in Github actions!

But for instantiating Prisma, I'm still not sure how Verce/Redner is accessing the prisma build files if we have them outside of the deployed folders. (i.e. for Vercel, we're deploying frontend/, and Vercel shouldn't be able to access files in db/.) Or is there a configuration that allows that? 
: ok so Vercel allows that?: https://vercel.com/docs/concepts/monorepos

**Comment by DanTehrani** [comment]
@cha0sg0d 
One way we can manage the migrations/Prisma scheme files in our local envs is to not only make sure with hooks that the Prisma schemas in all folders are equivalent but also create a hook to make sure all migration files are equivalent. I believe that will make things a little easier.

**Comment by cha0sg0d** [comment]
So regarding Vercel deployments, @amirgamil and I confirmed that deploys are working correctly with a monorepo. Vercel is smart enough to know that the installed `node_modules/` that the `frontend` needs are in the root of the repo. We also set the `DATABASE_URL` manually as a Vercel environment variable.

In the case of Prisma, after reading some docs + asking Chat GPT, when we run `prisma generate`, the type-safe Prisma client (which has our latest schema) is added to the `node_modules/.prisma/client` folder. So as long as we run the `prisma generate` from the `db` package before the build the `frontend`, the correct Prisma client will exist and `frontend` doens't have to directly interact with any files in the `db` package.

Re: Render, I'm not sure about how you all have been using render. But I created a stanadlone Postgres instance and was able to remotely run migrations via `prisma migrate deploy`. So it seems like Render is monorepo-agnostic in this case because it's just a database that we update via our CI.

Maybe we can chat more on the call on Monday to finalize the plan here.

**Review by DanTehrani — APPROVED** [review]
I think this PR is good to merge, I tested the pnpm commands and confirmed they work! Good commands!

File: packages/frontend/package.json [hunk 1]
```diff
@@ -3,16 +3,11 @@
   "version": "0.1.0",
   "private": true,
   "scripts": {
-    "build": "pnpm generate && next build",
+    "build": "next build",
```
Comment (amirgamil) [inline packages/frontend/package.json:6]: does prisma correctly generate types now with just `next build`?

Comment (cha0sg0d) [reply packages/frontend/package.json:6]: Right now the types are generated by the top level `prepare` script on the initial install.

`"prepare": "husky install && pnpm -F db run build && pnpm -F nymjs run build",`

I think we can add the `@personaelabs/db` as a dependency to the `frontend` and have a script to build the `Prisma client` in the `frontend/package.json` if needed. wdyt?

**Pull request merged**